src/AuthenticationService.php

```diff
@@ -215,6 +215,12 @@ public function clearIdentity(ServerRequestInterface $request, ResponseInterface
     {
         foreach ($this->authenticators() as $authenticator) {
             if ($authenticator instanceof PersistenceInterface) {
+                if ($authenticator->isImpersonating($request)) {
+                    /** @psalm-var array{request: \Cake\Http\ServerRequest, response: \Cake\Http\Response} $stopImpersonationResult */
+                    $stopImpersonationResult = $authenticator->stopImpersonating($request, $response);
+                    $request = $stopImpersonationResult['request'];
+                    $response = $stopImpersonationResult['response'];
```

Comment:
Should we use list destructuring instead? `[$request, $response] = $stopImpersonationResult;` ?